index.js

```diff
@@ -91,13 +102,19 @@ function checkAndReport(node, timeout) {
 					}
 					console.groupCollapsed('%c%s: %c%s %s', fmt, result.impact, defaultReset, result.help, result.helpUrl);
 					result.nodes.forEach(function (node) {
-						var el = document.querySelector(node.target.toString());
-						console.error(node.failureSummary);
-						console.log('HTML: %c%s', boldCourier, node.html);
-						if (!el) {
-							console.log('Selector: %c%s', boldCourier, node.target.toString());
-						} else {
-							console.log('Element: %o', el);
+						console.error(failureSummary(node));
+						logHtmlAndElement(node);
+
+						let allAnyNone = node.all.concat(node.any, node.none);
+
+						if (allAnyNone.length > 0) {
+							console.groupCollapsed('Related nodes');
+							allAnyNone.forEach(function (check) {
+								check.relatedNodes.forEach(function (relatedNode) {
```

Comment:
This is going to confuse users if multiple checks return related nodes. In that case, they will not be able to tell which related node belongs to which failed check. The related node information should be interspersed with the appropriate check messages. As you can see from this file https://github.com/dequelabs/axe-core/blob/master/lib/core/reporters/helpers/format-check.js each check is formatted on its own with its related nodes and then the entire thing is combined into the overall message for the node.